pkg/service/webhook/service.go

```diff
@@ -83,3 +163,56 @@ func (s Service) GetSupportedNouns() GetSupportedNounsResponse {
 func (s Service) GetSupportedVerbs() GetSupportedVerbsResponse {
 	return GetSupportedVerbsResponse{Verbs: []Verb{Create, Delete}}
 }
+
+func (s Service) PublishWebhook(noun Noun, verb Verb, payload interface{}) {
+	webhook, err := s.storage.GetWebhook(context.Background(), string(noun), string(verb))
+	if err != nil {
+		logrus.WithError(err).Warn("get webhook")
+	}
+
+	if webhook == nil {
+		return
+	}
+
+	for _, url := range webhook.URLS {
+		postPayload := Payload{Noun: noun, Verb: verb, URL: url, Data: payload}
+
+		postJSONData, err := json.Marshal(postPayload)
+		if err != nil {
+			logrus.Warn("marshal payload")
+		}
+		_, err = s.post(url, string(postJSONData))
+		if err != nil {
+			logrus.Warnf("posting payload to %s", url)
+		}
+	}
+}
+
+func (s Service) post(url string, json string) (string, error) {
+	req, err := http.NewRequest(http.MethodPost, url, bytes.NewBuffer([]byte(json)))
```

Comment:
I did this on purpose to not mess up the context for our DB and stuff.

If this hangs or something with the existing context and we hit unknown services, could this have an affect on our db/routes/timeouts?

Reply:
It wouldn't have an effect since: a) contexts are copied and; b) this is all running in a separate goroutine.

You could either pass in the same context (which ends up being copied), or create a new context with a new deadline (if you want to increase it).